pyomo/solvers/plugins/solvers/cplex_direct.py

```diff
@@ -906,9 +906,9 @@ def _warm_start(self):
             var_names = []
             var_values = []
             for pyomo_var, cplex_var in self._pyomo_var_to_solver_var_map.items():
-                if pyomo_var.value is not None:
+                if pyomo_var.value is not None and not (integer_only_warmstarts and pyomo_var.is_continuous()):
                     var_names.append(cplex_var)
-                    var_values.append(value(pyomo_var))
+                    var_values.append(value(pyomo_var) if pyomo_var.is_continuous() else round(value(pyomo_var)))
```

Comment:
I think ideally we want a unit test for this and for the contents of the MST file